homeassistant/components/nws/weather.py

```diff
@@ -140,96 +139,72 @@ def __init__(
         self._attr_name = self.station
 
     async def async_added_to_hass(self) -> None:
-        """Set up a listener and load data."""
+        """When entity is added to hass."""
         await super().async_added_to_hass()
-        self.async_on_remove(
-            self.coordinator_forecast_legacy.async_add_listener(
-                self._handle_legacy_forecast_coordinator_update
-            )
-        )
-        # Load initial data from coordinators
-        self._handle_coordinator_update()
-        self._handle_hourly_forecast_coordinator_update()
-        self._handle_twice_daily_forecast_coordinator_update()
-        self._handle_legacy_forecast_coordinator_update()
-
-    @callback
-    def _handle_coordinator_update(self) -> None:
-        """Load data from integration."""
-        self.observation = self.nws.observation
-        self.async_write_ha_state()
-
-    @callback
-    def _handle_hourly_forecast_coordinator_update(self) -> None:
-        """Handle updated data from the hourly forecast coordinator."""
-        self._forecast_hourly = self.nws.forecast_hourly
+        self.async_on_remove(partial(self._remove_forecast_listener, "daily"))
+        self.async_on_remove(partial(self._remove_forecast_listener, "hourly"))
+        self.async_on_remove(partial(self._remove_forecast_listener, "twice_daily"))
 
-    @callback
-    def _handle_twice_daily_forecast_coordinator_update(self) -> None:
-        """Handle updated data from the twice daily forecast coordinator."""
-        self._forecast_twice_daily = self.nws.forecast
+        if self.forecast_coordinators["twice_daily"] is not None:
+            self.unsub_forecast["twice_daily"] = self.forecast_coordinators[
+                "twice_daily"
+            ].async_add_listener(partial(self._handle_forecast_update, "twice_daily"))
 
-    @callback
-    def _handle_legacy_forecast_coordinator_update(self) -> None:
-        """Handle updated data from the legacy forecast coordinator."""
-        self._forecast_legacy = self.nws.forecast
-        self.async_write_ha_state()
+        if self.forecast_coordinators["hourly"] is not None:
+            self.unsub_forecast["hourly"] = self.forecast_coordinators[
+                "hourly"
+            ].async_add_listener(partial(self._handle_forecast_update, "hourly"))
 
     @property
     def native_temperature(self) -> float | None:
         """Return the current temperature."""
-        if self.observation:
-            return self.observation.get("temperature")
+        if self.nws.observation:
+            return self.nws.observation.get("temperature")
         return None
 
     @property
     def native_pressure(self) -> int | None:
         """Return the current pressure."""
-        if self.observation:
-            return self.observation.get("seaLevelPressure")
+        if self.nws.observation:
+            return self.nws.observation.get("seaLevelPressure")
```

Comment:
```suggestion
        if observation := self.nws.observation:
            return observation.get("seaLevelPressure")
```

Could be written like this to avoid the duplicate lookups

Reply:
Went through and change all `if self.nws.observation` like this.